pubsub/consumer.go

```diff
@@ -0,0 +1,152 @@
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/offchainlabs/nitro/util/redisutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/spf13/pflag"
+)
+
+type ConsumerConfig struct {
+	// Intervals in which consumer will update heartbeat.
+	KeepAliveInterval time.Duration `koanf:"keepalive-interval"`
+	// Duration after which consumer is considered to be dead if heartbeat
+	// is not updated.
+	KeepAliveTimeout time.Duration `koanf:"keepalive-timeout"`
```

Comment:
there is an inherent connection between KeepAliveTimout here (when the hearbeat disappears) and KeepAliveInterval (how often you update it). If you have both you should check that interval is shorter than timeout. Another option is to only have one config and always set interval to half timeout or vice-versa